src/native/managed/native-library.targets

```diff
@@ -0,0 +1,84 @@
+<Project>
+    <ItemGroup>
+            <PackageReference Include="Microsoft.DotNet.Build.Tasks.Templating" Version="$(MicrosoftDotNetBuildTasksTemplatingVersion)" />
+    </ItemGroup>
+
+    <!-- set a default include path, if there wasn't one already in the project -->
+    <ItemGroup>
+        <NativeLibraryCmakeFragmentIncludePath Condition="'@(NativeLibraryCmakeFragmentIncludePath->Count())' == '0' and Exists($([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)\..\inc')))" Include="$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)\..\inc'))" />
+    </ItemGroup>
+
+
+    <PropertyGroup>
+        <NativeLibraryCMakeFindPackageConfigOutputPath>$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)\cmake\find_package\'))</NativeLibraryCMakeFindPackageConfigOutputPath>
+    </PropertyGroup>
+
+
+    <!-- Write a ProjectName.cmake fragment into the publish artifact folder with that helps with consuming the NativeAOT produced library -->
+
+    <Target Name="WriteNativeLibraryCMakeFragment" DependsOnTargets="SetupProperties;WriteNativeLibraryCMakeFragmentCommon;WriteNativeLibraryCMakeFragmentShared" AfterTargets="Publish" Condition="'$(PublishAot)' == 'true'"
+            Outputs="$(NativeLibraryCMakeFragmentPath);$(NativeLibraryCMakeFindPackageConfigPath)">
+        <!-- Note: GenerateFileFromTemplate uses ${replacementVar} syntax for things that it will replace.
+             Use a backtick to write a normal cmake variable reference: `${nonReplacedVar}
+             Wish: GenerateFileFromTemplate should provide a way to specify the replacement sigil.
+        -->
```

Comment:
Open an issue for this to track improving it?

Reply:
https://github.com/dotnet/arcade/issues/14552